examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,189 @@
+ */
+
+#define ENOUGH 10 // FIXME: use correct value
+
+
+typedef struct fizzbuzz
+{
+    int input;
+    char *result;
+    qmlbind_signal_emitter *signalEmitter;
+} fizzbuzz_t;
+
+
+void on_changed(fizzbuzz_t *fb, const qmlbind_value *input)
+{
+    int i = (int) qmlbind_value_get_number(input);
+    if (i % 15 == 0) {
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputWasFizzBuzz",
+                                    0, NULL);
+        strcpy(fb->result, "FizzBuzz");
+    } else if (i % 3 == 0) {
+        strcpy(fb->result, "Fizz");
+    } else if (i % 5 == 0) {
+        strcpy(fb->result, "Buzz");
+    } else {
+        sprintf(fb->result, "%d", i);
+    }
+
+    const qmlbind_value *params[1] = {
+        qmlbind_value_new_string_cstr(fb->result)
+    };
+    qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1, params);
+    qmlbind_value_release(params[0]);
+}
+
+
+qmlbind_client_object *
+fizzbuzz_new_object_cb(qmlbind_client_class *classRef,
+                       qmlbind_signal_emitter *signalEmitter)
+{
+    fizzbuzz_t *fb = malloc(sizeof(fizzbuzz_t));
+    fb->input = 0;
+    fb->result = malloc(sizeof(char) * ENOUGH);
+    strcpy(fb->result, "");
+    fb->signalEmitter = signalEmitter;
+
+    return (qmlbind_client_object *) fb;
+}
+
+
+void fizzbuzz_delete_object_cb(qmlbind_client_object *obj)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+    free(fb->result);
+    qmlbind_signal_emitter_release(fb->signalEmitter);
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "quit") == 0) {
+        exit(0); // FIXME: should call app.quit()
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+qmlbind_value *fizzbuzz_get_property_cb(qmlbind_engine *engine,
+                                        qmlbind_client_object *obj,
+                                        const char *name)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input") == 0) {
+        return qmlbind_value_new_number(fb->input);
+    } else if (strcmp(name, "result") == 0) {
+        return qmlbind_value_new_string_cstr(fb->result);
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+void fizzbuzz_set_property_cb(qmlbind_engine *engine,
+                              qmlbind_client_object *obj,
+                              const char *name,
+                              const qmlbind_value *value)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input") == 0) {
+        fb->input = (int) qmlbind_value_get_number(value);
+
+        on_changed(fb, value);
+
+        const qmlbind_value *params[1] = {
```

Comment:
In cases where there's a `qmlbind_value` inside `params`, I think it's the best way to declare the values as non-const because the destructor wants a non-const pointer, but cast the array given to `qmlbind_signal_emitter_emit()` to `const qmlbind_value* const *`, similar as proposed for `argv` in the main function.